src/formats/chr.h

```diff
@@ -95,6 +95,14 @@ int sd_chr_load(sd_chr_file *chr, const char *filename);
  */
 int sd_chr_save(sd_chr_file *chr, const char *filename);
 
+/*! \brief Transform CHR pilot name to filename.
+ *
+ * \param chr CHR struct pointer to get pilot name from.
+ * \param filename Output parameter where the filename is written.
+ *                 This needs a buffer of size 9: 8 for letters and 1 for '\0'.
+ */
+void sd_chr_filename(const sd_chr_file *chr, char filename[static 9]);
```

Comment:
is there a reason sd_chr_filename doesn't append the `.CHR`?
and as per discussion in discord, let's simplify the `char filename[static 9]` parameter to something like a `char *filename` with a note in the doc comment stating how large a buffer is required (maybe provide a constant in the header, something like SD_CHR_FILENAME_LEN?).

Reply:
It's because the calling function is doing join with dirname and .CHR already, and the rename from old to normalized follow that pattern, so it would be messier to have .CHR here. About the 8 length, sure could put it as some constant somewhere.

Reply:
Yeah, need to have the old and new filename format have similar functions, so will most likely add the `.CHR` there, but I did add `MAX_CHR_BASENAME` in chr.h now. Anyway, this is currently broken if you for example create a pilot `A/B`, which becomes `A_B` and fails the first load on pilot creation.

Reply:
path injection in chr character names is less than ideal :sweat_smile:
perhaps skip the old-to-new-filename-renaming stuff if the character name contains `/` or `\`?

> Anyway, this is currently broken if you for example create a pilot `A/B`,

by currently, i assume you mean without this PR?

Reply:
Well the pilot name `A/B` is currently problematic with and without this pr, but I'll make it work. I rewrote this now with str but it's still broken in a couple places.